student_explorer/cron.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: future_fstrings -*-
+
+from __future__ import print_function #python 3 support
+
+from django.db import connections as conns
+
+import logging, datetime, json
+
+from django.conf import settings
+
+from django_cron import CronJobBase, Schedule
+
+from umich_api.api_utils import ApiUtil
+
+from seumich.models import Mentor
+
+import dateutil.parser as parser
+
+from decouple import config
+
+
+logger = logging.getLogger(__name__)
+
+# use ApiUtil for all API directory calls
+apiUtil = ApiUtil(config("APIUTIL_URL", default=""), config("APIUTIL_KEY", default=""), config("APIUTIL_SECRET", default=""))
+
+# cron job to populate course and user tables
+class StudentExplorerCronJob(CronJobBase):
+
+    schedule = Schedule(run_at_times=settings.RUN_AT_TIMES)
+    code = 'student_explorer.StudentExplorerCronJob'    # a unique code
+
+    def get_user_canvas_id(self, user_uniqname):
+        """
+        This process gets the canvas user id for user
+        :param user_uniqname:
+        :return:
+        """
+        user_result = apiUtil.api_call(f"aa/CanvasReadOnly/users/sis_login_id:{user_uniqname}/profile", "canvasreadonly")
+        user_json = json.loads(user_result.text)
+        return user_json.get("id", "")
+
+    def mentor_in_affiliated_department(self, mentor_uniqname):
+        """
+        checks whether the mentor is in affiliated department
+        :param mentor_uniqname:
+        :return:
+        """
+        in_affiliated_department = False
+
+        # get the array of affiliated department from config settings
+        deptAffiliations = config("DEPT_AFFILIATION", default="").split(",")
```

Comment:
This can be done by the library.
```suggestion
        deptAffiliations = config("DEPT_AFFILIATION", default="", cast=Csv())
```